Runtime/AvatarCreator/Scripts/UI/Buttons/SelectionButton.cs

```diff
@@ -27,10 +29,16 @@ public void AddListener(Action action)
         /// </summary>
         /// <param name="texture">The texture to be assigned to the RawImage component</param>
         /// <param name="sizeToParent">If true the icon will resize itself to fit inside the parent RectTransform</param>
-        public void SetIcon(Texture texture, bool sizeToParent = true)
+        public void SetIcon(Texture texture)
         {
+            if (rawImageRectTransform == null)
+            {
+                rawImageRectTransform = rawImage.GetComponent<RectTransform>();
+            }
```

Comment:
What about converting variable to property and defining this inside of the property:

```
private Rigidbody m_rigidbody = null;
 
private Rigidbody rigidbody {
   get {
     if (!m_rigidbody)
       m_rigidbody = GetComponent<Rigidbody>();
 
     return m_rigidbody;
   }
}
```

Reply:
It's not a bad idea, but I do think it is kinda overkill and makes the code a bit less readable. 
If we had other functions that made use of the RectTransform it would make more sense, but at the moment it is isolated only to the SetIcon function.

Reply:
got it, i don't have strong feelings about this so feel free to merge.